server/src/main/java/io/crate/planner/WhereClauseOptimizer.java

```diff
@@ -216,6 +222,25 @@ public static List<Integer> getPartitionIndices(List<ColumnIdent> pkCols, List<C
         return result;
     }
 
+    private static boolean queryHasPkSymbolsOnly(Symbol query, DocTableInfo table) {
+        var docKeyColumns = new ArrayList<>(table.primaryKey());
+        docKeyColumns.addAll(table.partitionedBy());
+        docKeyColumns.add(table.clusteredBy());
+        docKeyColumns.add(DocSysColumns.VERSION);
+        docKeyColumns.add(DocSysColumns.SEQ_NO);
+        docKeyColumns.add(DocSysColumns.PRIMARY_TERM);
+
+        boolean[] hasPkSymbolsOnly = new boolean[] {true};
+        Consumer<Reference> addRefIfMatch = ref -> {
```

Comment:
The name doesn't make sense,  it's not adding any refs?

Maybe just inline the consumer